lib/reservation_tracker.rb

```diff
@@ -0,0 +1,48 @@
+# handle availabilty 
+# handle rooms
+# room availability
+module Hotel 
+ 
+    class ReservationTracker
+        attr_reader :reservations, :all_rooms, :all_availabile, :block_reservation
+
+        def initialize(max_rooms)
+            if max_rooms > 20 
+                raise ArgumentError, "this hotel is too large, max 20 rooms"
+            else 
+                @all_rooms = [*1..max_rooms]
+            end
+            @reservations = []
+            @all_availabile = []
+        end
+
+#reservation methods 
+
+        def book_reservation(room = nil, checkin_date, checkout_date)
+            reservation = Hotel::Reservation.new(room, checkin_date, checkout_date)
+            @reservations << reservation
+            return reservation
+        end
+
+        def reservations_by_date(date)
+            list_all_rooms = []
+            @reservations.each do |reservation|
+                if date == reservation.checkin_date
+                    list_all_rooms << reservation
+                end
```

Comment:
Nice-- looking back, if you had more time, this `each` loop is a good opportunity to refactor to use an `Enumerable` method (look up `select`)